src/init.cpp

```diff
@@ -1741,7 +1742,8 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
             }
         } else {
             const std::string network_type = bind_arg.substr(index + 1);
-            if (network_type == "onion") {
+            // Do not fill in connOptions.onion_binds if you are not running the Tor service
+            if (args.GetBoolArg("-listenonion", DEFAULT_LISTEN_ONION) && network_type == "onion") {
```

Comment:
This would ignore `bind=...=onion` if `listenonion=0`. That will break the static Tor configurations where the hidden service is configured in the Tor daemon and `bitcoind` is running with `-bind=...=onion -listenonion=0`.

See `## 3. Manually create a Bitcoin Core onion service` in `doc/tor.md`.